src/test/compile-fail/privacy/restricted/tuple-struct-fields/test.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(pub_restricted)]
+
+macro_rules! id {
+    (e:$expr) => { $e }
+}
+
+mod foo {
+    type T = ();
+    struct S1(pub(foo) (), pub(T), pub(crate) (), pub(((), T)));
+    struct S2(pub(id!(foo)) ()); //~ ERROR expected one of `+` or `,`, found `(`
+                                 //~| ERROR expected one of `+`, `;`, or `where`, found `(`
```

Comment:
@nikomatsakis we have this test case, but the macro `id` should be
```rust
#![feature(type_macros)]
macro_rules! id {
    ($t:ty) => { $t }
}
```
instead of how it is defined now.

Is this the type of test you had in mind?